src/parser/jobs/mnk/Demolish.js

```diff
@@ -0,0 +1,100 @@
+import React, {Fragment} from 'react'
+
+import {ActionLink, StatusLink} from 'components/ui/DbLink'
+import ACTIONS from 'data/ACTIONS'
+import STATUSES from 'data/STATUSES'
+
+import Module from 'parser/core/Module'
+import {Rule, Requirement} from 'parser/core/modules/Checklist'
+import {TieredSuggestion, SEVERITY} from 'parser/core/modules/Suggestions'
+
+const DEMO_DURATION_MILLIS = STATUSES.DEMOLISH.duration * 1000
+
+export default class Demolish extends Module {
+	static handle = 'demolish'
+	static dependencies = [
+		'checklist',
+		'enemies',
+		'invuln',
+		'suggestions',
+	]
+
+	_clipDemo = 0
+	_lastDemo = {}
+
+	constructor(...args) {
+		super(...args)
+
+		const filter = {
+			by: 'player',
+			abilityId: STATUSES.DEMOLISH.id,
+		}
+
+		this.addHook(['applydebuff', 'refreshdebuff'], filter, this._onApply)
+		this.addHook('complete', this._onComplete)
+	}
+
+	// Aggregate Demo uptime
+	_onApply(event) {
+		const lastDemo = this._lastDemo[event.targetID] = this._lastDemo[event.targetID] || 0
+
+		if (!lastDemo) {
+			this._lastDemo[event.targetID] = event.timestamp
+			return
+		}
+
+		let clip = DEMO_DURATION_MILLIS - (event.timestamp - lastDemo)
+
+		clip -= this.invuln.getUntargetableUptime('all', event.timestamp - DEMO_DURATION_MILLIS, event.timestamp)
+		clip -= this.invuln.getInvulnerableUptime('all', event.timestamp, event.timestamp + DEMO_DURATION_MILLIS + clip)
```

Comment:
lookaheads :notlikeblob:

remind me to research how the hell boss specific invulns are gonna work because i have noooooo idea